interop_binaries/tests/end_to_end.rs

```diff
@@ -0,0 +1,598 @@
+        ))
+        .header(CONTENT_TYPE, JSON_MEDIA_TYPE)
+        .json(&json!({
+            "taskId": task_id_encoded,
+            "aggregatorId": 0,
+            "hostnameAndPort": format!("127.0.0.1:{}", leader_port),
+        }))
+        .send()
+        .await?;
+    assert_eq!(leader_endpoint_response.status(), StatusCode::OK);
+    assert_eq!(
+        leader_endpoint_response
+            .headers()
+            .get(CONTENT_TYPE)
+            .unwrap(),
+        JSON_MEDIA_TYPE,
+    );
+    let leader_endpoint_response_body = leader_endpoint_response.json::<Value>().await?;
+    let leader_endpoint_response_object = leader_endpoint_response_body
+        .as_object()
+        .context("endpoint_for_task response is not an object")?;
+    assert_eq!(
+        leader_endpoint_response_object
+            .get("status")
+            .context("endpoint_for_task response is missing \"status\"")?,
+        "success",
+        "error: {:?}",
+        leader_endpoint_response_object.get("error"),
+    );
+    assert_eq!(
+        leader_endpoint_response_object
+            .get("endpoint")
+            .context("endpoint_for_task response is missing \"endpoint\"")?,
+        "/",
+    );
+
+    // Send a /internal/test/endpoint_for_task request to the helper.
+    let helper_endpoint_response = http_client
+        .post(format!(
+            "http://127.0.0.1:{}/internal/test/endpoint_for_task",
+            helper_port,
+        ))
+        .header(CONTENT_TYPE, JSON_MEDIA_TYPE)
+        .json(&json!({
+            "taskId": task_id_encoded,
+            "aggregatorId": 1,
+            "hostnameAndPort": format!("127.0.0.1:{}", leader_port),
+        }))
+        .send()
+        .await?;
+    assert_eq!(helper_endpoint_response.status(), StatusCode::OK);
+    assert_eq!(
+        helper_endpoint_response
+            .headers()
+            .get(CONTENT_TYPE)
+            .unwrap(),
+        JSON_MEDIA_TYPE,
+    );
+    let helper_endpoint_response_body = helper_endpoint_response.json::<Value>().await?;
+    let helper_endpoint_response_object = helper_endpoint_response_body
+        .as_object()
+        .context("endpoint_for_task response is not an object")?;
+    assert_eq!(
+        helper_endpoint_response_object
+            .get("status")
+            .context("endpoint_for_task response is missing \"status\"")?,
+        "success",
+        "error: {:?}",
+        helper_endpoint_response_object.get("error"),
+    );
+    assert_eq!(
+        helper_endpoint_response_object
+            .get("endpoint")
+            .context("endpoint_for_task response is missing \"endpoint\"")?,
+        "/",
+    );
+
+    // Send a /internal/test/add_task request to the collector.
+    let collector_add_task_response = http_client
+        .post(format!(
+            "http://127.0.0.1:{}/internal/test/add_task",
+            collector_port,
+        ))
+        .header(CONTENT_TYPE, JSON_MEDIA_TYPE)
+        .json(&json!({
```

Comment:
```suggestion
        .json(&json!({
```
[`RequestBuilder::json` automatically sets `Content-Type: application/json`](https://docs.rs/reqwest/latest/src/reqwest/async_impl/request.rs.html#441-457) so I think quite a few calls to `RequestBuilder::header` can be deleted.